manifests/repo.pp

```diff
@@ -0,0 +1,12 @@
+# Configure the foreman repo
+class foreman::repo(
+) {
+  if ! $::foreman::custom_repo {
```

Comment:
I think these should all be class parameters (defaulting to ```$::foreman::*```) so it can be used without the main foreman class, even if you have to manually set all parameters.